tests/python/contrib/test_ethosn/infrastructure.py

```diff
@@ -0,0 +1,225 @@
+            "relay.ext.ethos-n.options": {"variant": 0}
+    }):
+        with tvm.target.create("llvm -mcpu=core-avx2"):
+            if npu:
+                f = relay.build_module.bind_params_by_name(mod["main"], params)
+                mod = tvm.IRModule()
+                mod["main"] = f
+                mod = relay.transform.AnnotateTarget("ethos-n")(mod)
+                mod = relay.transform.MergeCompilerRegions()(mod)
+                mod = relay.transform.PartitionGraph()(mod)
+                cpu_op_count = get_cpu_op_count(mod)
+                assert cpu_op_count == cpu_ops, \
+                    "Got {} CPU operators, expected {}".format(cpu_op_count, cpu_ops)
+                partition_count = 0
+                for global_var in mod.get_global_vars():
+                    if "ethos-n" in global_var.name_hint:
+                        partition_count += 1
+
+                assert npu_partitions == partition_count, \
+                    "Got {} ethos-n partitions, expected {}".format(partition_count, npu_partitions)
+
+            return relay.build(mod, params=params)
+
+
+def run(graph, lib, params, inputs, outputs, npu=True):
+    module = graph_runtime.create(graph, lib, tvm.cpu())
+    module.set_input(**inputs)
+    module.set_input(**params)
+    module.run()
+    out = [module.get_output(i) for i in range(outputs)]
+    if not npu:
+        inference_result(0, out)
+    return out
+
+
+def build_and_run(mod, inputs, outputs, params, ctx=tvm.cpu(), npu=True, cpu_ops=0, npu_partitions=1):
+    graph, lib, params = build(mod, params, npu, cpu_ops, npu_partitions)
+    return run(graph, lib, params, inputs, outputs, npu)
+
+
+def verify(answers, atol, rtol=1e-07, verify_saturation=True):
+    """Compare the array of answers. Each entry is a list of outputs"""
+    if len(answers) < 2:
+        print("No results to compare: expected at least two, found ",
+              len(answers))
+    for answer in zip_longest(*answers):
+        for outs in combinations(answer, 2):
+            if verify_saturation:
+                assert np.count_nonzero(outs[0].asnumpy() == 255) < 0.25 * outs[0].asnumpy().size, \
+                    "Output is saturated: {}".format(outs[0])
+                assert np.count_nonzero(outs[0].asnumpy() == 0) < 0.25 * outs[0].asnumpy().size, \
+                    "Output is saturated: {}".format(outs[0])
+            tvm.testing.assert_allclose(
+                outs[0].asnumpy(), outs[1].asnumpy(), rtol=rtol, atol=atol
+            )
+
+
+def inference_result(checksum, outputs):
+    """Set the expected results of an Ethos inference, if the testing
+    infrastructure is available. This assumes that the entire graph
+    was offloaded to the neural processor."""
+    if tvm.get_global_func(
+            "relay.ethos-n.test.infra.inference_result", True):
+        return _infrastructure.inference_result(checksum, *outputs)
+    return False
+
+
+def generate_trials(space, r_factor=3):
```

Comment:
Although this function is not being used, I'd provide the same suggestion as I did to the ACL integration. We should avoid using the random workloads in the unit test.

Reply:
Moved into 2nd PR, can discuss further there.